app/controllers/annotations_controller.rb

```diff
@@ -2,110 +2,40 @@ class AnnotationsController < ApplicationController
-        example_answer = init_annotation(params[:example_answer_text], @question, current_user.org, Annotation.types[:example_answer])
-      end
-    end
-    # only required to save if we updated/created one
-    ex_save = example_answer.present? ? example_answer.save : true
-    guid_save = guidance.present? ? guidance.save : true
+    question = Question.includes(section: { phase: :template}).find(params[:question_id])
+    authorize question
 
-    @section = @question.section
-    @phase = @section.phase
-    @phase.template.dirty = true
+#    example_answer = Annotation.find_or_create_by(question: question, org: current_user.org, type: Annotation.types[:example_answer])
+#    guidance = Annotation.find_or_create_by(question: question, org: current_user.org, type: Annotation.types[:guidance])
+    
+    hash = {question_id: question.id, org_id: current_user.org.id}
+    process_changes(hash.merge({type: Annotation.types[:example_answer]}), params[:example_answer_text], _('example answer'))
+    process_changes(hash.merge({type: Annotation.types[:guidance]}), params[:guidance_text], _('guidance'))
 
-    if ex_save && guid_save
-      typ = (example_answer.present? && guidance.present? ? 'example answer and guidance' : (guidance.present? ? 'guidance' : 'example answer'))
-      redirect_to admin_show_phase_path(id: @phase.id, section_id: @section.id, question_id: @question.id, edit: 'true'), notice: success_message(typ, _('saved'))
-    else
-      if !ex_save && !guid_save
-        flash[:alert] = failed_create_error(example_answer, _('example answer')) + '\n' +
-                          failed_create_error(gudiance, _('guidance'))
-      elsif !guid_save
-        flash[:alert] = failed_create_error(gudiance, _('guidance'))
-      elsif !ex_save
-        flash[:alert] = failed_create_error(example_answer, _('example answer'))
-      end
-      render action: "phases/admin_show"
+    if !flash[:notice].blank? || !flash[:alert].blank?
+      template = question.section.phase.template
+      template.dirty = true
+      template.save!
     end
+    redirect_to controller: :phases, action: :admin_show, id: question.section.phase.id
   end
 
   #delete an annotation
   def admin_destroy
-    @example_answer = Annotation.includes(question: { section: {phase: :template}}).find(params[:id])
-    authorize @example_answer
-    @question = @example_answer.question
-    @section = @question.section
-    @phase = @section.phase
-    @phase.template.dirty = true
-    if @example_answer.destroy
-      redirect_to admin_show_phase_path(id: @phase.id, section_id: @section.id, edit: 'true'), notice: success_message(_('information'), _('deleted'))
-    else
-      redirect_to admin_show_phase_path(id: @phase.id, section_id: @section.id, edit: 'true'), notice: flash[:alert] = failed_destroy_error(@example_answer, _('example answer'))
+    annotation = Annotation.find(params[:id])
+    authorize annotation
+    phase_id = annotation.question.section.phase.id
```

Comment:
with this change still needs to run 4 queries. I would propose this:
```
Annotation.joins("INNER JOIN questions ON annotations.question_id = questions.id").joins("INNER JOIN sections ON questions.section_id = sections.id").joins("INNER JOIN phases ON sections.phase_id = phases.id").where("annotations.id": params[:id]).pluck("phases.id").first
```
although it is quite verbose, just loads the phase.id and since we redirect to admin_show anyway...